go/stats/counter_test.go

```diff
@@ -71,3 +71,31 @@ func TestGaugeFunc(t *testing.T) {
 		t.Errorf("want 1, got %v", v.String())
 	}
 }
+
+func TestGaugeFloat64(t *testing.T) {
+	var gotname string
+	var gotv *GaugeFloat64
+	clear()
+	Register(func(name string, v expvar.Var) {
+		gotname = name
+		gotv = v.(*GaugeFloat64)
+	})
+	v := NewGaugeFloat64("f", "help")
+	if gotname != "f" {
```

Comment:
nit: I personally find stretchr assertions like `assert.Equal(t, 3.14, v.Get())` much easier to read